pandas/_libs/groupby_helper.pxi.in

```diff
@@ -19,6 +19,18 @@ ctypedef fused rank_t:
     object
 
 
+cdef inline bint _treat_as_na(rank_t val, bint is_datetimelike) nogil:
+    if rank_t is object:
```

Comment:
Somewhat counter-intuitive that this works - out of curiosity what was the complaint?

Reply:
AFAICT the `val != val` check requires calling `val.__ne__` which requires the gil

Reply:
shouldn’t need the gil if these are c level object (eg ints or floats)

Reply:
Right, we just need to exclude the object case from getting to the `val == val` step, thats all this check is doing